src/components/CompactServer/CompactServer.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import {ICompactServerProps, ServerStatus} from './CompactServer.Props';
+import {TagContainer} from '../TagContainer/TagContainer';
+import { Icon } from '../Icon/Icon';
+import * as classNames from 'classnames';
+import  {autobind} from '../../utilities/autobind';
+import'./CompactServer.scss';
+
+function checkFilter(filter: string, serverName: string) {
+    return serverName.toLowerCase().trim().indexOf(filter.toLowerCase().trim()) !== -1;
+}
+
+export class CompactServer extends React.Component<ICompactServerProps, any> {
+
+    constructor(props?: ICompactServerProps) {
+        super(props);
+    }
+
+    getRoleDisplay(role) {
+        return role.display;
+    }
+    
+     public shouldComponentUpdate(nextProps: ICompactServerProps, nextState) {
+        return !(this.props.serverName === nextProps.serverName
+            && this.props.status === nextProps.status
+            && this.props.roleList === nextProps.roleList);
+    }
+
+    render() {
+         let isCritical = this.props.status === ServerStatus.Critical;
+         let isWarning = this.props.status === ServerStatus.Warning;
+         let isOK = this.props.status === ServerStatus.OK;
+         
+
+        let name = this.props.serverName;
+        if (name.length > 15) {

```

Comment:
Yes :D There were some problems, but I've managed to get it right. Will be in the next commit.